### PR TITLE
MAUI example for Super Resolution.

### PR DESCRIPTION
**Pull request by skottmckay**
Uses RealESRGAN which is a signficantly larger model but has significantly better output than the simple pytorch model we have been using. Pre/post processing is different though so including script to add that.

This could be a template for building out example models to keep things modular and simple - i.e. example app + script that acquires model and defines the full PrePostProcessor pipeline locally. The predefined helpers in extensions are somewhat model specific anyway and it's a little indirect to use a pipeline defined there.

### CODE REVIEW

File: mobile/examples/super_resolution/MAUI/real_esrgan_ort_e2e.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    so = ort.SessionOptions()
+    so.graph_optimization_level = ort.GraphOptimizationLevel.ORT_ENABLE_EXTENDED
+
+    # register the custom operators for the image decode/encode pre/post processing  provided by onnxruntime-extensions
+    # with onnxruntime. if we do not do this we'll get an error on model load about the operators not being found.
+    ortext_lib_path = get_library_path()
+    so.register_custom_ops_library(ortext_lib_path)
+    inference_session = ort.InferenceSession(model_path, so)
+
+    test_image_path = _this_dirpath / 'inputs' / 'lr_lion.png'
```
Comment (edgchen1) [inline mobile/examples/super_resolution/MAUI/real_esrgan_ort_e2e.py:106]:
```suggestion
    test_image_path = _this_dirpath / 'MauiSuperResolution' / 'Resources' / 'Raw' / 'lr_lion.png'
```

File: mobile/examples/super_resolution/MAUI/MauiSuperResolution/OrtInferenceSession.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        XNNPACK // XNNPACK is available on ARM/ARM64 platforms and benefits 32-bit float models
+    }
+
+    // An inference session runs an ONNX model
+    class OrtInferenceSession
+    {
+        public OrtInferenceSession(ExecutionProviders provider = ExecutionProviders.CPU)
+        {
+            _sessionOptions = new SessionOptions();
+            switch (_executionProvider)
```
Comment (edgchen1) [inline mobile/examples/super_resolution/MAUI/MauiSuperResolution/OrtInferenceSession.cs:21]:
```suggestion
            switch (provider)
```
maybe we don't need `_executionProvider`

File: mobile/examples/super_resolution/MAUI/Nuget.config [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <packageSources>
+    <clear />
+    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
+    <add key="ORT-Nightly" value="https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ORT-Nightly/nuget/v3/index.json" />
```
Comment (edgchen1) [inline mobile/examples/super_resolution/MAUI/Nuget.config:6]: could we use an official release or is a nightly build required?

**Review by edgchen1 — APPROVED** [review]

File: mobile/examples/super_resolution/MAUI/MauiSuperResolution/Platforms/Android/AndroidManifest.xml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android">
+	<application android:allowBackup="true" android:icon="@mipmap/appicon" android:roundIcon="@mipmap/appicon_round" android:supportsRtl="true"></application>
+	<uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
+	<uses-permission android:name="android.permission.INTERNET" />
```
Comment (edgchen1) [inline mobile/examples/super_resolution/MAUI/MauiSuperResolution/Platforms/Android/AndroidManifest.xml:5]: are these permissions required?

File: mobile/examples/super_resolution/MAUI/LICENSE.txt [hunk 5]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
[the remark this answers is not in the document]
Comment (skottmckay) [reply mobile/examples/super_resolution/MAUI/LICENSE.txt:1]: Removed.

File: mobile/examples/super_resolution/MAUI/MauiSuperResolution/Platforms/Android/AndroidManifest.xml (hunk 4 above)
Comment (skottmckay) [reply mobile/examples/super_resolution/MAUI/MauiSuperResolution/Platforms/Android/AndroidManifest.xml:5]: I don't think so. Remnants from some unrelated testing. Removed.

**Pull request merged**